steam/ext/dota2/models.py

```diff
@@ -0,0 +1,197 @@
+    game_time:
+        Timedelta representing in-game timer.
+        Draft stage and time before runes (0:00 mark) have this as non-positive timedelta.
+    delay
+        Time delay for the match if watching in the Dota 2 application.
+        Similarly, the data in attributes of this class is also behind the present by this delay.
+    last_update_time:
+        Time the data was updated last by the Game Coordinator.
+    tournament
+        Tournament information, if the match is a tournament game.
+    radiant_lead
+        Amount of gold lead Radiant team has. Negative value in case Dire is leading.
+    radiant_score
+        Amount of kills Radiant team has
+    dire_score
+        Amount of kills Dire team has
+    building_state
+        Bitmask. An integer that represents a binary of which buildings are still standing.
+    custom_game_difficulty
+        Custom Game Difficulty
+    """
+
+    def __init__(self, state: GCState, proto: CSourceTVGameSmall) -> None:
+        self._state: GCState = state
+
+        self.match_id: int = proto.match_id
+        self.server_steam_id: int = proto.match_id
+        self.lobby_id: int = proto.lobby_id
+
+        self.lobby_type = CSODOTALobbyLobbyType.try_value(proto.lobby_type)
+        self.game_mode = DOTAGameMode.try_value(proto.game_mode)
+        self.average_mmr: int = proto.average_mmr
+        self.sort_score: int = proto.sort_score
+        self.spectators: int = proto.spectators
+
+        self.start_time = datetime.datetime.fromtimestamp(proto.activate_time, datetime.timezone.utc)
+        self.game_time = datetime.timedelta(seconds=proto.game_time)
+        self.end_time = datetime.timedelta(seconds=proto.deactivate_time)
+        self.delay = datetime.timedelta(seconds=proto.delay)
+        self.last_update_time = datetime.datetime.fromtimestamp(proto.last_update_time, datetime.timezone.utc)
+
+        self.tournament: TournamentMatch | None = None
+        if proto.league_id:  # if it is 0 then all tournament related fields are going to be 0 as well
+            battle_cup = None
+            # todo: check if battle cup has league_id, otherwise we need to separate tournament from battle_cup
+            if proto.weekend_tourney_tournament_id:  # if it is 0 then all battle cup related fields are going to be 0
+                battle_cup = BattleCup(
+                    proto.weekend_tourney_tournament_id,
+                    proto.weekend_tourney_division,
+                    proto.weekend_tourney_skill_level,
+                    proto.weekend_tourney_bracket_round,
+                )
+            self.tournament = TournamentMatch(
+                proto.league_id,
+                proto.series_id,
+                [
+                    TournamentTeam(proto.team_id_radiant, proto.team_name_radiant, proto.team_logo_radiant),
+                    TournamentTeam(proto.team_id_dire, proto.team_name_dire, proto.team_logo_dire),
+                ],
```

Comment:
Should this be a tuple if it's fixed length?

Reply:
At first, thought about it but csgo `.teams` is typehinted as `: list[Team]` so I went with same... Isn't it also 2 teams in csgo? I'm not a player of it.